pkg/controller/erspan.go

```diff
@@ -0,0 +1,358 @@
+		"name":      podIf.ObjectMeta.Name,
+	})
+}
+
+func erspanInit(cont *AciController, stopCh <-chan struct{}) {
+	cont.log.Debug("Initializing erspan client")
+	restconfig := cont.env.RESTConfig()
+	erspanClient, err := erspanclientset.NewForConfig(restconfig)
+	if err != nil {
+		cont.log.Errorf("Failed to intialize erspan client")
+		return
+	}
+	cont.initErspanInformerFromClient(erspanClient)
+	cont.erspanInformer.Run(stopCh)
+}
+
+func podIfInit(cont *AciController, stopCh <-chan struct{}) {
+	cont.log.Debug("Initializing podIf client")
+	restconfig := cont.env.RESTConfig()
+	podIfClient, err := podIfclientset.NewForConfig(restconfig)
+	if err != nil {
+		cont.log.Errorf("Failed to intialize podif client")
+		return
+	}
+	cont.initPodIfInformerFromClient(podIfClient)
+	cont.podIfInformer.Run(stopCh)
+}
+
+func (cont *AciController) initErspanInformerFromClient(
+	erspanClient *erspanclientset.Clientset) {
+	cont.initErspanInformerBase(
+		cache.NewListWatchFromClient(
+			erspanClient.AciV1alpha().RESTClient(), "erspanpolicies",
+			metav1.NamespaceAll, fields.Everything()))
+}
+
+func (cont *AciController) initPodIfInformerFromClient(
+	podIfClient *podIfclientset.Clientset) {
+	cont.initPodIfInformerBase(
+		cache.NewListWatchFromClient(
+			podIfClient.AciV1().RESTClient(), "podifs",
+			metav1.NamespaceAll, fields.Everything()))
+}
+
+func (cont *AciController) initErspanInformerBase(listWatch *cache.ListWatch) {
+	cont.erspanIndexer, cont.erspanInformer = cache.NewIndexerInformer(
+		listWatch, &erspanpolicy.ErspanPolicy{}, 0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				cont.erspanPolicyUpdated(obj)
+			},
+			UpdateFunc: func(_, obj interface{}) {
+				cont.erspanPolicyUpdated(obj)
+			},
+			DeleteFunc: func(obj interface{}) {
+				cont.erspanPolicyDelete(obj)
+			},
+		},
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+	)
+	cont.log.Debug("Initializing erspan Policy Informers")
+}
+
+func (cont *AciController) initPodIfInformerBase(listWatch *cache.ListWatch) {
+	cont.podIfIndexer, cont.podIfInformer = cache.NewIndexerInformer(
+		listWatch, &podIfpolicy.PodIF{}, 0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				cont.podIFAdded(obj)
+			},
+			UpdateFunc: func(oldobj interface{}, newobj interface{}) {
+				cont.podIFAdded(newobj)
+			},
+			DeleteFunc: func(obj interface{}) {
+				cont.podIFDeleted(obj)
+			},
+		},
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+	)
+	cont.log.Debug("Initializing podif Informers")
+}
+
+func getPodifKey(podif *podIfpolicy.PodIF) string {
+	return podif.Status.PodNS + "/" + podif.Status.PodName
+}
+
+func (cont *AciController) podIFAdded(obj interface{}) {
+	podif := obj.(*podIfpolicy.PodIF)
+	cont.log.Infof("podif Added: %s", podif.ObjectMeta.Name)
+	cont.indexMutex.Lock()
+	deconcatenate := strings.Split(podif.Status.EPG, "|")
+	deconEPG := deconcatenate[len(deconcatenate)-1]
```

Comment:
Although it shouldn't happen, we should add some error checking here in case the EPG name isn't of the expected form to avoid a runtime error